lib/knight.rb

```diff
@@ -0,0 +1,41 @@
+class Knight
+  attr_reader :x, :y
+  VECTORS = [[-2, -1], [-2, 1], [-1, -2], [-1, 2], [1, -2], [1, 2], [2, -1], [2, 1]].freeze
+  BOARD_RANGE = (1..8)
+
+  def initialize(position)
+    @x, @y = to_coordinate(position)
+  end
+
+  def possible_moves
+    moves.select { |move| on_board?(move) }.map { |move| to_position(move) }
+  end
+
+  private
+
+  def to_position(coordinates)
+    (coordinates.first + 96).chr << coordinates.last.to_s
+  end
+
+  def to_coordinate(position)
+    [position[0].ord - 96, position[1].to_i]
+  end
+
+  def on_board?(move)
+    BOARD_RANGE.cover?(move.first) && BOARD_RANGE.cover?(move.last)
+  end
+
+  def moves
+    coords = []
+    VECTORS.map do |coordinates|
+      coords << [x + coordinates.first, y + coordinates.last]
+    end
+    coords
+  end
+end
+
+unless ENV['RACK_ENV'] == 'rspec'
+  File.open(ARGV[0]).each_line do |line|
```

Comment:
in Knight class you can 

```ruby
def to_s 
   possible_moves.sort.join(' ')
end
```

and in this loop you could simplify this to 

```ruby
  puts Knight.new(line)
```

this is part of duck typing (you expect method to implement `to_s` so you can display the text representation of the object on the screen